lxd/instance/drivers/driver_lxc.go

```diff
@@ -4205,6 +4205,13 @@ func (d *lxc) Update(args db.InstanceArgs, userRequested bool) error {
 		return newDevType.UpdatableFields(oldDevType)
 	})
 
+	// Prevent changing device initial configuration properties.
+	for _, key := range allUpdatedKeys {
```

Comment:
I wonder if we should allow removing the initial keys, this way it would allow an instance to be prepares to be exported for use in an older version of LXD.

But changing or adding new initial keys should be blocked.

Reply:
To me, removing initial configuration seems like a viable option.

I've also tested if removing initial configuration makes any unexpected effect on migrations, and can confirm that this is safe.

Will improve the PR.

Reply:
Follow up - validation for container works as expected (just some minor fixes are required in tests). 

However, initial configuration disturbs the evacuation process of the VM (both if VM is running or stopped).
```
Error: Failed to migrate instance "vm1" in project "default": Instance move to destination failed: Error transferring instance data: Failed migration on target: Error reading migration control source: websocket: close 1000 (normal)
``` 
^ I've tested without initial configuration and it works as expected.

In addition, removing initial configuration from the VM's config (only if VM is running) results in:
```
$ lxc config device unset vm2 root initial.zfs.block_mode

Error: Failed to start device "root": Failed to add drive config: Invalid source path "": stat : no such file or directory
```

I'm still investigating this, but just wanted to point it out to see if you have some clues/hints what is going on?

Reply:
> `Error: Failed to migrate instance "vm1" in project "default": Instance move to destination failed: Error transferring instance data: Failed migration on target: Error reading migration control source: websocket: close 1000 (normal)`

For this check `lxc monitor --debug` on the target machine and check if there is a more useful error.

Reply:
I suspect that this is linked to the #12203. After rebase, everything is working as expected besides when `initial.zfs.block_mode` is set to `true` the *websocket closed* error appears. If the setting is set to `false`, migration works as expected.

Reply:
The actual error is "Failed migration on source                    clusterMoveSourceName=vm1 err="Filesystem zvol detected in source but target does not support receiving zvols" instance=vm1 live=false project=default push=false" 

Here is a complete log:
```
DEBUG  [2023-09-14T13:00:54Z] New operation                                 class=task description="Evacuating cluster member" operation=2d3bc9b7-1167-4d8a-bfe5-30504ee665ef project=
DEBUG  [2023-09-14T13:00:54Z] Started operation                             class=task description="Evacuating cluster member" operation=2d3bc9b7-1167-4d8a-bfe5-30504ee665ef project=
INFO   [2023-09-14T13:00:54Z] ID: 2d3bc9b7-1167-4d8a-bfe5-30504ee665ef, Class: task, Description: Evacuating cluster member  CreatedAt="2023-09-14 13:00:54.911755004 +0000 UTC" Err= Location=cls-1 MayCancel=false Metadata="map[]" Resources="map[]" Status=Pending StatusCode=Pending UpdatedAt="2023-09-14 13:00:54.911755004 +0000 UTC"
INFO   [2023-09-14T13:00:54Z] ID: 2d3bc9b7-1167-4d8a-bfe5-30504ee665ef, Class: task, Description: Evacuating cluster member  CreatedAt="2023-09-14 13:00:54.911755004 +0000 UTC" Err= Location=cls-1 MayCancel=false Metadata="map[]" Resources="map[]" Status=Running StatusCode=Running UpdatedAt="2023-09-14 13:00:54.911755004 +0000 UTC"
DEBUG  [2023-09-14T13:00:54Z] Handling API request                          ip=@ method=GET protocol=unix url=/1.0/operations/2d3bc9b7-1167-4d8a-bfe5-30504ee665ef username=root
DEBUG  [2023-09-14T13:00:55Z] Shutdown request sent to instance             instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:00:55Z] Updated metadata for operation                class=task description="Evacuating cluster member" operation=2d3bc9b7-1167-4d8a-bfe5-30504ee665ef project=
INFO   [2023-09-14T13:00:55Z] ID: 2d3bc9b7-1167-4d8a-bfe5-30504ee665ef, Class: task, Description: Evacuating cluster member  CreatedAt="2023-09-14 13:00:54.911755004 +0000 UTC" Err= Location=cls-1 MayCancel=false Metadata="map[evacuation_progress:Stopping \"vm1\" in project \"default\"]" Resources="map[]" Status=Running StatusCode=Running UpdatedAt="2023-09-14 13:00:55.003419499 +0000 UTC"
DEBUG  [2023-09-14T13:00:55Z] Instance operation lock created               action=stop instance=vm1 project=default reusable=true
DEBUG  [2023-09-14T13:00:55Z] Shutdown started                              instance=vm1 instanceType=virtual-machine project=default timeout=30s

// bunch of successful heartbeats ...

DEBUG  [2023-09-14T13:01:19Z] Instance stopped                              instance=vm1 instanceType=virtual-machine project=default reason=guest-shutdown target=stop
DEBUG  [2023-09-14T13:01:19Z] onStop hook started                           instance=vm1 instanceType=virtual-machine project=default target=stop
DEBUG  [2023-09-14T13:01:19Z] Instance operation lock inherited for stop    action=stop instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:19Z] Waiting for VM process to finish              instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:19Z] QMP monitor stopped                           path=/var/snap/lxd/common/lxd/logs/vm1/qemu.monitor
DEBUG  [2023-09-14T13:01:20Z] VM process finished                           instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:20Z] Stopping device                               device=root instance=vm1 instanceType=virtual-machine project=default type=disk
DEBUG  [2023-09-14T13:01:20Z] UnmountInstance started                       driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:20Z] Unmounted ZFS dataset                         dev=default/virtual-machines/vm1 driver=zfs path=/var/snap/lxd/common/lxd/storage-pools/default/virtual-machines/vm1 pool=default volName=vm1
DEBUG  [2023-09-14T13:01:20Z] Deactivated ZFS volume                        dev=default/virtual-machines/vm1.block driver=zfs pool=default volName=vm1
DEBUG  [2023-09-14T13:01:20Z] UnmountInstance finished                      driver=zfs instance=vm1 pool=default project=default
INFO   [2023-09-14T13:01:20Z] Action: instance-shutdown, Source: /1.0/instances/vm1 
DEBUG  [2023-09-14T13:01:20Z] Instance operation lock finished              action=stop err="<nil>" instance=vm1 project=default reusable=true
DEBUG  [2023-09-14T13:01:20Z] Shutdown finished                             instance=vm1 instanceType=virtual-machine project=default timeout=30s
DEBUG  [2023-09-14T13:01:20Z] onStop hook finished                          instance=vm1 instanceType=virtual-machine project=default target=stop
DEBUG  [2023-09-14T13:01:20Z] Updated metadata for operation                class=task description="Evacuating cluster member" operation=2d3bc9b7-1167-4d8a-bfe5-30504ee665ef project=
INFO   [2023-09-14T13:01:20Z] ID: 2d3bc9b7-1167-4d8a-bfe5-30504ee665ef, Class: task, Description: Evacuating cluster member  CreatedAt="2023-09-14 13:00:54.911755004 +0000 UTC" Err= Location=cls-1 MayCancel=false Metadata="map[evacuation_progress:Migrating \"vm1\" in project \"default\" to \"cls-2\"]" Resources="map[]" Status=Running StatusCode=Running UpdatedAt="2023-09-14 13:01:20.364978948 +0000 UTC"
DEBUG  [2023-09-14T13:01:20Z] Connecting to a remote LXD over HTTPS         url="https://10.8.100.120:8443"
DEBUG  [2023-09-14T13:01:20Z] New operation                                 class=websocket description="Migrating instance" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
INFO   [2023-09-14T13:01:20Z] ID: c2d3ec39-d131-4286-90e0-9764f79b8fa3, Class: websocket, Description: Migrating instance  CreatedAt="2023-09-14 13:01:20.406666805 +0000 UTC" Err= Location=cls-1 MayCancel=true Metadata="map[control:61953de3859717407c601c67613729a5fe343a1084b42db8bd36c06725a29272 fs:a1ec2edd99078aea7fca19e2ab9adc5e924a44043dc7be4ccd3e20f5d8282c9d]" Resources="map[instances:[/1.0/instances/vm1]]" Status=Pending StatusCode=Pending UpdatedAt="2023-09-14 13:01:20.406666805 +0000 UTC"
INFO   [2023-09-14T13:01:20Z] ID: c2d3ec39-d131-4286-90e0-9764f79b8fa3, Class: websocket, Description: Migrating instance  CreatedAt="2023-09-14 13:01:20.406666805 +0000 UTC" Err= Location=cls-1 MayCancel=true Metadata="map[control:61953de3859717407c601c67613729a5fe343a1084b42db8bd36c06725a29272 fs:a1ec2edd99078aea7fca19e2ab9adc5e924a44043dc7be4ccd3e20f5d8282c9d]" Resources="map[instances:[/1.0/instances/vm1]]" Status=Running StatusCode=Running UpdatedAt="2023-09-14 13:01:20.406666805 +0000 UTC"
DEBUG  [2023-09-14T13:01:20Z] Started operation                             class=websocket description="Migrating instance" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
INFO   [2023-09-14T13:01:20Z] Waiting for migration control connection on source  clusterMoveSourceName=vm1 instance=vm1 live=false project=default push=false
DEBUG  [2023-09-14T13:01:20Z] Connected to the websocket: wss://10.8.100.120:8443/1.0/events?project=default&target=cls-2 
DEBUG  [2023-09-14T13:01:20Z] Sending request to LXD                        etag= method=POST url="https://10.8.100.120:8443/1.0/instances?project=default&target=cls-2"
DEBUG  [2023-09-14T13:01:20Z] 
        {
                "architecture": "x86_64",
                "config": {
                        "image.architecture": "amd64",
                        "image.description": "ubuntu 22.04 LTS amd64 (release) (20230914)",
                        "image.label": "release",
                        "image.os": "ubuntu",
                        "image.release": "jammy",
                        "image.serial": "20230914",
                        "image.type": "disk-kvm.img",
                        "image.version": "22.04",
                        "volatile.base_image": "1e5c8a4627a4121bbde2749e4e6fbb313dba3052e2053c3f3f624e0543724cba",
                        "volatile.cloud-init.instance-id": "5e6bb666-10d7-489f-bb7d-6af708507d21",
                        "volatile.evacuate.origin": "cls-1",
                        "volatile.last_state.power": "RUNNING",
                        "volatile.uuid": "05cbcc8d-f434-48fa-a568-2139f064676c",
                        "volatile.uuid.generation": "05cbcc8d-f434-48fa-a568-2139f064676c",
                        "volatile.vsock_id": "3549251101"
                },
                "devices": {
                        "root": {
                                "initial.zfs.block_mode": "true",
                                "path": "/",
                                "pool": "default",
                                "type": "disk"
                        }
                },
                "ephemeral": false,
                "profiles": [],
                "stateful": false,
                "description": "",
                "name": "vm1",
                "source": {
                        "type": "migration",
                        "certificate": "-----BEGIN CERTIFICATE-----\nMIIB4TCCAWegAwIBAgIRAOwyGq0fE542eypgvXr+HwQwCgYIKoZIzj0EAwMwIzEM\nMAoGA1UEChMDTFhEMRMwEQYDVQQDDApyb290QGNscy0xMB4XDTIzMDkxNDExMjAy\nMloXDTMzMDkxMTExMjAyMlowIzEMMAoGA1UEChMDTFhEMRMwEQYDVQQDDApyb290\nQGNscy0xMHYwEAYHKoZIzj0CAQYFK4EEACIDYgAEh1QXlcEOH47L1qvAzPWvPWel\nFrcpSBNDO/A6V60HL/314U9KBv9f7lZJmKXfEvJfzOx1pfhW9wQNcSZkhjmHATGJ\nCHXBqytnsLj+grru+UKR6m+WP+Sw1nXwhRKghCTxo18wXTAOBgNVHQ8BAf8EBAMC\nBaAwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYDVR0TAQH/BAIwADAoBgNVHREEITAf\nggVjbHMtMYcEfwAAAYcQAAAAAAAAAAAAAAAAAAAAATAKBggqhkjOPQQDAwNoADBl\nAjEAmQID4/g89BSamcsqfSqDuUFKYTUPUa63cfvP3paVUObCFZSfocBogU3WIbt3\nfX04AjAzch+1TqoWcbvfP5Acx4ld8C/XvuaASaQPG1wo9MQ4bD/aQvnwcgk7jpLj\nuuQYlqM=\n-----END CERTIFICATE-----\n",
                        "mode": "pull",
                        "operation": "https://10.8.100.244:8443/1.0/operations/c2d3ec39-d131-4286-90e0-9764f79b8fa3",
                        "secrets": {
                                "control": "61953de3859717407c601c67613729a5fe343a1084b42db8bd36c06725a29272",
                                "fs": "a1ec2edd99078aea7fca19e2ab9adc5e924a44043dc7be4ccd3e20f5d8282c9d"
                        },
                        "source": "vm1",
                        "allow_inconsistent": false
                },
                "instance_type": "",
                "type": "virtual-machine"
        } 
INFO   [2023-09-14T13:01:20Z] ID: 148be2d5-59c8-4035-9b17-a3ec36365628, Class: task, Description: Creating instance  CreatedAt="2023-09-14 13:01:20.531796364 +0000 UTC" Err= Location=cls-2 MayCancel=false Metadata="map[]" Resources="map[instances:[/1.0/instances/vm1]]" Status=Running StatusCode=Running UpdatedAt="2023-09-14 13:01:20.531796364 +0000 UTC"
DEBUG  [2023-09-14T13:01:20Z] Got operation from LXD                       
DEBUG  [2023-09-14T13:01:20Z] Sending request to LXD                        etag= method=GET url="https://10.8.100.120:8443/1.0/operations/148be2d5-59c8-4035-9b17-a3ec36365628?project=default&target=cls-2"
DEBUG  [2023-09-14T13:01:20Z] 
        {
                "id": "148be2d5-59c8-4035-9b17-a3ec36365628",
                "class": "task",
                "description": "Creating instance",
                "created_at": "2023-09-14T13:01:20.531796364Z",
                "updated_at": "2023-09-14T13:01:20.531796364Z",
                "status": "Running",
                "status_code": 103,
                "resources": {
                        "instances": [
                                "/1.0/instances/vm1"
                        ]
                },
                "metadata": null,
                "may_cancel": false,
                "err": "",
                "location": "cls-2"
        } 
INFO   [2023-09-14T13:01:20Z] ID: 148be2d5-59c8-4035-9b17-a3ec36365628, Class: task, Description: Creating instance  CreatedAt="2023-09-14 13:01:20.531796364 +0000 UTC" Err= Location=cls-2 MayCancel=false Metadata="map[]" Resources="map[instances:[/1.0/instances/vm1]]" Status=Pending StatusCode=Pending UpdatedAt="2023-09-14 13:01:20.531796364 +0000 UTC"
DEBUG  [2023-09-14T13:01:20Z] Allowing untrusted GET                        ip="10.8.100.120:58234" url="/1.0/operations/c2d3ec39-d131-4286-90e0-9764f79b8fa3/websocket?secret=61953de3859717407c601c67613729a5fe343a1084b42db8bd36c06725a29272"
DEBUG  [2023-09-14T13:01:20Z] Connecting to operation                       class=websocket description="Migrating instance" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
DEBUG  [2023-09-14T13:01:20Z] Connected to operation                        class=websocket description="Migrating instance" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
INFO   [2023-09-14T13:01:20Z] Migration send starting                       instance=vm1 instanceType=virtual-machine project=default
INFO   [2023-09-14T13:01:20Z] Migration control connection established on source  clusterMoveSourceName=vm1 instance=vm1 live=false project=default push=false
DEBUG  [2023-09-14T13:01:20Z] Got response struct from LXD                 
DEBUG  [2023-09-14T13:01:20Z] 
        {
                "id": "148be2d5-59c8-4035-9b17-a3ec36365628",
                "class": "task",
                "description": "Creating instance",
                "created_at": "2023-09-14T13:01:20.531796364Z",
                "updated_at": "2023-09-14T13:01:20.531796364Z",
                "status": "Running",
                "status_code": 103,
                "resources": {
                        "instances": [
                                "/1.0/instances/vm1"
                        ]
                },
                "metadata": null,
                "may_cancel": false,
                "err": "",
                "location": "cls-2"
        } 
DEBUG  [2023-09-14T13:01:20Z] Allowing untrusted GET                        ip="10.8.100.120:58244" url="/1.0/operations/c2d3ec39-d131-4286-90e0-9764f79b8fa3/websocket?secret=a1ec2edd99078aea7fca19e2ab9adc5e924a44043dc7be4ccd3e20f5d8282c9d"
DEBUG  [2023-09-14T13:01:20Z] Connecting to operation                       class=websocket description="Migrating instance" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
DEBUG  [2023-09-14T13:01:20Z] Connected to operation                        class=websocket description="Migrating instance" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
DEBUG  [2023-09-14T13:01:20Z] MountInstance started                         driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:21Z] Activated ZFS volume                          dev=default/virtual-machines/vm1.block driver=zfs pool=default volName=vm1
DEBUG  [2023-09-14T13:01:21Z] MountInstance finished                        driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:21Z] Mounted ZFS dataset                           dev=default/virtual-machines/vm1 driver=zfs path=/var/snap/lxd/common/lxd/storage-pools/default/virtual-machines/vm1 pool=default volName=vm1
DEBUG  [2023-09-14T13:01:21Z] UnmountInstance started                       driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:21Z] Unmounted ZFS dataset                         dev=default/virtual-machines/vm1 driver=zfs path=/var/snap/lxd/common/lxd/storage-pools/default/virtual-machines/vm1 pool=default volName=vm1
DEBUG  [2023-09-14T13:01:21Z] UnmountInstance finished                      driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:21Z] Deactivated ZFS volume                        dev=default/virtual-machines/vm1.block driver=zfs pool=default volName=vm1
DEBUG  [2023-09-14T13:01:21Z] Set migration offer volume size               blockSize=10737418240 instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Sending migration offer to target             instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Waiting for migration offer response from target  instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Migrate send control monitor started          instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Got migration offer response from target      instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Migrate send transfer started                 instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] MigrateInstance started                       args="&{IndexHeaderVersion:1 Name:vm1 Snapshots:[] MigrationType:{FSType:ZFS Features:[migration_header compress]} TrackProgress:true MultiSync:false FinalSync:false Data:<nil> ContentType: AllowInconsistent:false Refresh:false Info:0xc000011da8 VolumeOnly:false ClusterMove:true}" driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:21Z] Sent migration index header, waiting for response  args="&{IndexHeaderVersion:1 Name:vm1 Snapshots:[] MigrationType:{FSType:ZFS Features:[migration_header compress]} TrackProgress:true MultiSync:false FinalSync:false Data:<nil> ContentType: AllowInconsistent:false Refresh:false Info:0xc000011da8 VolumeOnly:false ClusterMove:true}" driver=zfs instance=vm1 pool=default project=default version=1
DEBUG  [2023-09-14T13:01:21Z] Websocket: Sending barrier message            address="10.8.100.120:58244"
DEBUG  [2023-09-14T13:01:21Z] Websocket: Got barrier message                address="10.8.100.120:58244"
INFO   [2023-09-14T13:01:21Z] Received migration index header response      args="&{IndexHeaderVersion:1 Name:vm1 Snapshots:[] MigrationType:{FSType:ZFS Features:[migration_header compress]} TrackProgress:true MultiSync:false FinalSync:false Data:<nil> ContentType: AllowInconsistent:false Refresh:false Info:0xc000011da8 VolumeOnly:false ClusterMove:true}" driver=zfs instance=vm1 pool=default project=default response="{StatusCode:200 Error: Refresh:0xc002be453b}" version=1
DEBUG  [2023-09-14T13:01:21Z] Websocket: Sending barrier message            address="10.8.100.120:58244"
DEBUG  [2023-09-14T13:01:21Z] MigrateInstance finished                      args="&{IndexHeaderVersion:1 Name:vm1 Snapshots:[] MigrationType:{FSType:ZFS Features:[migration_header compress]} TrackProgress:true MultiSync:false FinalSync:false Data:<nil> ContentType: AllowInconsistent:false Refresh:false Info:0xc000011da8 VolumeOnly:false ClusterMove:true}" driver=zfs instance=vm1 pool=default project=default
DEBUG  [2023-09-14T13:01:21Z] Migrate send transfer finished                instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Migrate send control monitor finished         instance=vm1 instanceType=virtual-machine project=default
INFO   [2023-09-14T13:01:21Z] Migration send stopped                        instance=vm1 instanceType=virtual-machine project=default
ERROR  [2023-09-14T13:01:21Z] Failed migration on source                    clusterMoveSourceName=vm1 err="Filesystem zvol detected in source but target does not support receiving zvols" instance=vm1 live=false project=default push=false
INFO   [2023-09-14T13:01:21Z] Migration channels disconnected on source     clusterMoveSourceName=vm1 instance=vm1 live=false project=default push=false
DEBUG  [2023-09-14T13:01:21Z] Failure for operation                         class=websocket description="Migrating instance" err="Failed migration on source: Filesystem zvol detected in source but target does not support receiving zvols" operation=c2d3ec39-d131-4286-90e0-9764f79b8fa3 project=default
INFO   [2023-09-14T13:01:21Z] ID: c2d3ec39-d131-4286-90e0-9764f79b8fa3, Class: websocket, Description: Migrating instance  CreatedAt="2023-09-14 13:01:20.406666805 +0000 UTC" Err="Failed migration on source: Filesystem zvol detected in source but target does not support receiving zvols" Location=cls-1 MayCancel=false Metadata="map[control:61953de3859717407c601c67613729a5fe343a1084b42db8bd36c06725a29272 fs:a1ec2edd99078aea7fca19e2ab9adc5e924a44043dc7be4ccd3e20f5d8282c9d]" Resources="map[instances:[/1.0/instances/vm1]]" Status=Failure StatusCode=Failure UpdatedAt="2023-09-14 13:01:20.406666805 +0000 UTC"
WARNING[2023-09-14T13:01:21Z] Failed sending migration failure to source    err="websocket: close sent" instance=vm1 instanceType=virtual-machine project=default
DEBUG  [2023-09-14T13:01:21Z] Updated metadata for operation                class=task description="Evacuating cluster member" operation=2d3bc9b7-1167-4d8a-bfe5-30504ee665ef project=
INFO   [2023-09-14T13:01:21Z] ID: 2d3bc9b7-1167-4d8a-bfe5-30504ee665ef, Class: task, Description: Evacuating cluster member  CreatedAt="2023-09-14 13:00:54.911755004 +0000 UTC" Err= Location=cls-1 MayCancel=false Metadata="map[]" Resources="map[]" Status=Running StatusCode=Running UpdatedAt="2023-09-14 13:01:21.517621607 +0000 UTC"
ERROR  [2023-09-14T13:01:21Z] Failed migration on target                    clusterMoveSourceName=vm1 err="Error reading migration control source: websocket: close 1000 (normal)" instance=vm1 live=false project=default push=false
INFO   [2023-09-14T13:01:21Z] ID: 148be2d5-59c8-4035-9b17-a3ec36365628, Class: task, Description: Creating instance  CreatedAt="2023-09-14 13:01:20.531796364 +0000 UTC" Err="Error transferring instance data: Failed migration on target: Error reading migration control source: websocket: close 1000 (normal)" Location=cls-2 MayCancel=false Metadata="map[]" Resources="map[instances:[/1.0/instances/vm1]]" Status=Failure StatusCode=Failure UpdatedAt="2023-09-14 13:01:20.531796364 +0000 UTC"
DEBUG  [2023-09-14T13:01:21Z] Failure for operation                         class=task description="Evacuating cluster member" err="Failed to migrate instance \"vm1\" in project \"default\": Instance move to destination failed: Error transferring instance data: Failed migration on target: Error reading migration control source: websocket: close 1000 (normal)" operation=2d3bc9b7-1167-4d8a-bfe5-30504ee665ef project=
INFO   [2023-09-14T13:01:21Z] ID: 2d3bc9b7-1167-4d8a-bfe5-30504ee665ef, Class: task, Description: Evacuating cluster member  CreatedAt="2023-09-14 13:00:54.911755004 +0000 UTC" Err="Failed to migrate instance \"vm1\" in project \"default\": Instance move to destination failed: Error transferring instance data: Failed migration on target: Error reading migration control source: websocket: close 1000 (normal)" Location=cls-1 MayCancel=false Metadata="map[]" Resources="map[]" Status=Failure StatusCode=Failure UpdatedAt="2023-09-14 13:01:21.517621607 +0000 UTC"
DEBUG  [2023-09-14T13:01:21Z] Event listener server handler stopped         listener=748bdbf2-6354-43d5-9fcd-07f34fca7e6b local=/var/snap/lxd/common/lxd/unix.socket remote=@
```

Reply:
What was the outcome of this? Did you check whether the target was indeed capable of creating zvol block mode volumes?

Reply:
All instances are the same, so I assume all of them have the same capabilities.
In addition this error does not occur for containers when using `zfs.block_mode` (which should if zvols are not supported?).

However, I do not really understand why `zfs.block_mode` would even have effect on VMs, since they are using two volumes anyway (one block and one FS volume)?